src/main/java/io/fabric8/maven/docker/StopMojo.java

```diff
@@ -105,4 +112,27 @@ private boolean invokedTogetherWithDockerStart() {
         Boolean startCalled = (Boolean) getPluginContext().get(CONTEXT_KEY_START_CALLED);
         return startCalled != null && startCalled;
     }
+
+    private Set<Network> getUnusedCustomNetworks(QueryService queryService, PomLabel pomLabel) throws DockerAccessException {
+        if (!createCustomNetworks) {
+            return Collections.emptySet();
+        }
+        Set<Network> customNetworks = new HashSet<>();
+        for (ImageConfiguration image : getImages()) {
+            final NetworkingMode networkingMode = image.getRunConfiguration().getNetworkingMode();
+            if (networkingMode.isCustomNetwork()) {
+                Network network = queryService.getNetworkByName(networkingMode.getCustomNetwork());
+                if (network != null) {
+                    customNetworks.add(network);
+                    for (Container container : getContainersToStop(queryService, image)) {
+                        if (!shouldStopContainer(container, pomLabel)) {
+                            // it's sill in use don't collect it
+                            customNetworks.remove(network);
+                        }
+                    }
+                }
+            }
+        }
+        return customNetworks;
+    }
```

Comment:
What's about the following situation:

* User creates custom network outside of Maven 
* User references this custom network in d-m-p image configs
* User calls 'mvn docker:stop' (without 'docker:start' in the same run but maybe in an extra call before)

At the end I guess the user probably expects the custom network to still exist. But if I understand the code correctly the network get removed nevertheless. 

For the container we had the same problem and could solve it by putting a d-m-p specific label on the container that were created by d-m-p itself. 

I wonder how we could solve the situation here. One could be to prefix all networks created by dmp with e.g. `dmp-` transparently so that it could be detected whether the network was created by the plugin or outside. 

wdyt ?